doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,860 @@
+3. ArsOrch binds to events from RouteOrch related to the newly created NHG.
+4. Notification
+    * RouteOrch component will notify ArsOrch of NHG update.
+3. Nexthop group's SAI attributes handling:
+    * If nexthop group's ARS object ID, "set" capability is supported:
+        - If all the routes that point to this nexthop group are ARS-enabled:
+            * The nexthop group is updated with the ARS object ID.
+        - else:
+            - Create a new nexthop group with the ARS object ID.
+            - Redirect ARS-enabled routes to use new nexthop group
+    * If "set" capability is not supported - implement via "create" functionality:
+        - Create a new nexthop group with the ARS object ID.
+        - Redirect ARS-enabled routes to use new nexthop group
+        - Remove original nexthop group if no route is pointing to it.
+    * If an alternative path is defined
+        - Set the relevant nexthops as part of the alternative path.
+
+__Figure 7: Nexthop group creation flow__
+![](images/nhg_create.png)
+
+1. Nexthop group creation
+    * RouteOrch receives route from APPL_DB, triggering the process.
+    * Nexthop group is created
+2. Notification
+    * RouteOrch component will notify ArsOrch of NHG update.
+3. Nexthop group's SAI attributes handling:
+    * If nexthop group's ARS object ID, "set" capability is supported:
+        - If all the routes that point to this nexthop group are ARS-enabled:
+            * The nexthop group is updated with the ARS object ID.
+        - else:
+            - Create a new nexthop group with the ARS object ID.
+            - Redirect ARS-enabled routes to use new nexthop group
+    * If "set" capability is not supported - implement via "create" functionality:
+        - Create a new nexthop group with the ARS object ID.
+        - Redirect ARS-enabled routes to use new nexthop group
+        - Remove original nexthop group if no route is pointing to it.
+    * If an alternative path is defined
+        - Set the relevant nexthops as part of the alternative path.
+
+__Figure 8: Nexthop group update flow__
+![](images/nhg_update.png)
+
+1. Nexthop group update
+    * RouteOrch receives route update from APPL_DB, triggering the process.
+    * Nexthop group is updated with new member
+2. Notification
+    * RouteOrch component will notify ArsOrch of NHG update.
+4. Nexthop group's SAI attributes handling:
+    * If an alternative path is defined
+        - Set the relevant nexthops as part of the alternative path.
+
+__Figure 9:ARS LAG table creation flow__
+![](images/ars_lag_create.png)
+
+1. Add ARS_PORTCHANNEL Table
+    * The CONFIG_DB adds the ARS_PORTCHANNEL table, triggering the process.
+2. Create SAI ARS Object
+    * The ArsOrch component receives the update and creates the SAI ARS object.
+4. LAG's SAI attributes handling:
+    * If LAG was created:
+        - It is updated with the ARS object ID.
+
+__Figure 10:LAG creation flow__

```

Comment:
Consider moving the support for LAG (SAI_LAG_ATTR_ARS_OBJECT_ID) to phase 2. This is supported by IFCS but not in TL SAI since it was not needed by customers so far.